tests/mocks/database/population/nonclassevents.population.ts

```diff
@@ -3,86 +3,75 @@ import { InjectRepository } from '@nestjs/typeorm';
 import { BasePopulationService } from './base.population';
-import { NonClassParentData, NonClassEventData } from './data';
+import { NonClassMeetingData } from './data';
+import { Room } from '../../../../src/server/location/room.entity';
 
 export class NonClassEventPopulationService
-  extends BasePopulationService<NonClassEvent> {
-  @InjectRepository(NonClassEvent)
-  protected eventRepository: Repository<NonClassEvent>;
-
+  extends BasePopulationService<NonClassParent> {
   @InjectRepository(NonClassParent)
   protected parentRepository: Repository<NonClassParent>;
 
+  @InjectRepository(NonClassEvent)
+  protected eventRepository: Repository<NonClassEvent>;
+
   @InjectRepository(Semester)
   protected semesterRepository: Repository<Semester>;
 
   @InjectRepository(Meeting)
   protected meetingRepository: Repository<Meeting>;
 
+  @InjectRepository(Room)
+  protected roomRepository: Repository<Room>;
+
   public async populate({
-    parents,
-    events,
-  }: {
-    parents: NonClassParentData[];
-    events: NonClassEventData[];
-  }): Promise<NonClassEvent[]> {
-    const [fall, spring] = await this.semesterRepository.find({
-      // The usage of an array of objects means typeorm will compile this
-      // into
-      // ((academicyear = 2019 AND term = "fall") OR  (academicYear = 2020 AND term = "spring"))
-      where: [
-        { academicYear: 2019, term: TERM.FALL },
-        { academicYear: 2020, term: TERM.SPRING },
-      ],
+    nonClassMeetings,
+  }: { nonClassMeetings: NonClassMeetingData[] }): Promise<NonClassParent[]> {
+    const allSemesters = await this.semesterRepository.find({
       order: {
         academicYear: 'ASC',
+        term: 'ASC',
       },
     });
 
-    const [
-      dataScienceParent,
-      appliedMathParent,
-    ] = await this.parentRepository.save([
+    const allRooms = await this.roomRepository.find(
       {
-        ...parents[0],
-      },
-      {
-        ...parents[1],
-      },
-    ]);
+        order: {
+          name: 'ASC',
+        },
+        relations: ['building'],
+      }
+    );
 
-    const meetings = await this.meetingRepository.find({
-      take: 4,
-    });
-    return this.eventRepository.save([
-      {
-        ...events[0],
-        nonClassParent: dataScienceParent,
-        semester: fall,
-        meetings: [meetings[0]],
-      },
-      {
-        ...events[0],
-        nonClassParent: dataScienceParent,
-        semester: spring,
-        meetings: [meetings[1]],
-      },
-      {
-        ...events[1],
-        nonClassParent: appliedMathParent,
-        semester: fall,
-        meetings: [meetings[2]],
-      },
-      {
-        ...events[1],
-        nonClassParent: appliedMathParent,
-        semester: spring,
-        meetings: [meetings[3]],
-      },
-    ]);
+    return this.parentRepository.save(
+      nonClassMeetings.map((parentData) => {
+        const nonClassParent = new NonClassParent();
```

Comment:
Is there a benefit to doing that over calling `new`? I'm not opposed, but all of the population services are using `new` right now and I'd like them to be consistent, so if we do that here I'd want to do that in all of them...

Reply:
I forget the specifics, but `new Entity()` can cause issues with missing fields (i.e: requiring `id` to be set etc.) it's been a while since I ran into that, so if it's working here then let's just leave it and revisit it later if we need to.